app/models/test.rb

```diff
@@ -4,8 +4,14 @@ class Test < ApplicationRecord
   has_many :questions, dependent: :destroy
   has_many :results, dependent: :destroy
   has_many :users, through: :results
-    def self.sort_body_test_desc(title)
-      joins(:category).where(categories: { title: :title })
-        .order(title: :desc).pluck(:title)
-    end
+
+  scope :light_test, -> { where(complexity: 0..1) }
+  scope :middle_test, -> { where(complexity: 2..4) }
+  scope :hard_test, -> { where(complexity: 5..Float::INFINITY) }
+
+  scope :sort_body_test_desc, -> (title) { joins(:category).where(categories: { title: :title }).order(title: :desc) }
+
+  validates :title, presence: true
+  validates :complexity, numericality: { only_integer: true, greater_than_or_equal_to: 1 }
+  validates_uniqueness_of :title, scope: :complexity
```

Comment:
этоту валидацию можно было совместить с проверкой на наличие
можно несколько валидаций разом указать вот так:
```ruby
validates :title, presence: true, uniqueness: ...
```

Reply:
валидации рекомендуют писать перед скоупами

Reply:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)